### PR TITLE
fix and improve generated documentation

### PR DESCRIPTION
**Pull request by ndeloof**

### CODE REVIEW

File: plugin/src/main/java/io/jenkins/plugins/casc/Configurator.java [hunk 1]
```diff
@@ -158,4 +158,10 @@ default Class getImplementedAPI() {
     @CheckForNull
     CNode describe(T instance, ConfigurationContext context) throws Exception;
 
+    /**
+     * Jelly hack used to distinguish RootElementConfigurator from other Configurators
+     */
+    default boolean isRootElement() {
+        return this instanceof RootElementConfigurator;
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/Configurator.java]: hacks are okay when they are this useful!

File: plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java [hunk 2]
```diff
@@ -143,7 +145,9 @@ protected Attribute detectActualType(String name, final TypePair type) {
         }
 
         Attribute attribute;
-        if (PersistedList.class.isAssignableFrom(type.rawType)) {
+        if (DescribableList.class.isAssignableFrom(type.rawType)) {
+            return new DescribableListAttribute(name, c);
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: So now we can get a list of attributes to show all the configurable elements?

File: plugin/src/main/java/io/jenkins/plugins/casc/Configurator.java (hunk 1 above)
Comment (ndeloof) [reply plugin/src/main/java/io/jenkins/plugins/casc/Configurator.java]: was for another fix in progress, I mixed my branches.

File: plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java (hunk 2 above)
Comment (ndeloof) [reply plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: capture d ecran 2018-10-10 a 10 35 35

Comment (casz) [reply plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: 🙌 and LINKS working too? 🙊

Comment (ndeloof) [reply plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: yes, sounds like we reach some milestone here :) 🎉

**Pull request merged**